### PR TITLE
Add solution method to nvti.

### PR DESCRIPTION
**Pull request by janowagner**
The solution method is handled analog to the solution type
and allows for a more refined formal solution definition.

### CODE REVIEW

File: base/nvti.c [hunk 1]
```diff
@@ -1203,6 +1219,28 @@ nvti_set_solution_type (nvti_t *n, const gchar *solution_type)
...
+ *
+ * @return 0 for success. Anything else indicates an error.
+ */
+int
+nvti_set_solution_method (nvti_t *n, const gchar *solution_method)
+{
+  if (!n)
+    return (-1);
+
+  if (n->solution_method)
```
Comment (mattmundell) [inline base/nvti.c]: There's no need to check this, because g_free does it.

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ### Added
 - Add nvti_get_tag() [#285](https://github.com/greenbone/gvm-libs/pull/285)
+- Add nvti_olution_metod() and nvti_set_solution_method() [#283](https://github.com/greenbone/gvm-libs/pull/283)
```
Comment (mattmundell) [inline CHANGELOG.md]:
```suggestion
- Add nvti_solution_metod() and nvti_set_solution_method() [#283](https://github.com/greenbone/gvm-libs/pull/283)
```

File: base/nvti.c (hunk 1 above)
Comment (janowagner) [reply base/nvti.c]: Done. Will fix the same for other cases in a separate patch.

File: CHANGELOG.md (hunk 2 above)
Comment (janowagner) [reply CHANGELOG.md]: Fixed.

File: base/nvti.c (hunk 1 above)
Comment (mattmundell) [reply base/nvti.c]: Thanks.

File: base/nvti_tests.c [hunk 3]
```diff
@@ -137,6 +153,7 @@ main (int argc, char **argv)
   add_test_with_context (suite, nvti, nvti_new_never_returns_null);
 
   add_test_with_context (suite, nvti, nvti_get_tag_gets_correct_value_one_tag);
+  add_test_with_context (suite, nvti, nvti_set_solution_method_correct);
```
Comment (mattmundell) [inline base/nvti_tests.c]: This should be on a fresh line below the nvti_get_tag tests.

Comment (janowagner) [reply base/nvti_tests.c]: done.

**Pull request merged**